New version: SHTns v0.3.0

JuliaRegistrator opened the pull request:
- Registering package: SHTns
- Repository: https://github.com/fgerick/SHTns.jl
- Created by: @fgerick
- Version: v0.3.0
- Commit: ce0285edd477585f3b790fa3332c3b00ce9a0ab2
- Reviewed by: @fgerick
- Reference: https://github.com/fgerick/SHTns.jl/commit/ce0285edd477585f3b790fa3332c3b00ce9a0ab2#commitcomment-155496597
- Release notes:
<!-- BEGIN RELEASE NOTES -->
`````
## Breaking changes

- Added GPU support, therefore change in `SHTnsType` definition.
- Change `SHTnsCfg` to include `Real` or `Complex` transform in type, instead of having both supported in one config. Avoids unnecessary ambiguities.
`````
<!-- END RELEASE NOTES -->

<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd600fbbc2cba1a143073ad916eafe864ae23f0263c13557d8d0900f43547c043bdbc87c51d53137d6dc0102b0cb30f3a03e12fb6d5b2c76aeb3af7b8c5364cc3a9d7365ece5d088f787eff22d601a811725630f3345eef9be25c91da5847e6132df5445fd07cb5a875abe115a385617d782e74ba2e84858f882ac6dc3caa5c76ae4337bbd15577f2cb27137bb0f32cc984167fc718d386aaa58f2b9aabe82d8eaa980bb0c1857ee0dadf63719f0d8cd576 -->

fgerick commented:
@JuliaRegistrator register

Release notes:

## Breaking changes

- Added GPU support, therefore change in `SHTnsType` definition.
- Change `SHTnsCfg` to include `Real` or `Complex` transform in type, instead of having both supported in one config. Avoids unnecessary ambiguities.

JuliaRegistrator committed:
New version: SHTns v0.3.0

UUID: 5dc4ea12-f621-4038-8458-a22eae100703
Repo: https://github.com/fgerick/SHTns.jl.git
Tree: 7cabdf036792c4832019c2bfc7d1be3f0d2d122c

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a
